### PR TITLE
[dev] summarisation example with HF Integration

### PR DESCRIPTION
**Pull request by kshitij12345**
Summarisation Demo

Run Example: https://app.neptune.ai/o/common/org/Kshiteej-test-bed/e/KSHIT-354

Custom-Dashboard : https://app.neptune.ai/o/common/org/Kshiteej-test-bed/e/KSHIT-354/dashboard/Custom-Dashboard-972adc81-a327-49d5-aec6-5bb060f9398d

main training file: `run_summarization.py`

### CODE REVIEW

**Comment by kshitij12345** [comment]
@SiddhantSadangi have updated the code structure. PTAL :)

File: notebooks/project-metadata.ipynb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import neptune.new as neptune\n",
+    "project = neptune.init_project(name=\"showcase/project-text-summarization-hf\")\n"
```
[the remark this answers is not in the document]
Comment (kshitij12345) [reply notebooks/project-metadata.ipynb]: As discussed offline, we should stick to `project-text-summarization-hf` as the code relies heavily on the HF Integration (and most of the meta-data logged comes from it).

File: summarization/hf_transformers/model/arg_parsers.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
+from dataclasses import dataclass, field
```
Comment (SiddhantSadangi) [inline summarization/hf_transformers/model/arg_parsers.py:1]: this file should be in the `scripts` folder

File: summarization/hf_transformers/model/run_summarization.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
+# File inspired from
```
Comment (SiddhantSadangi) [inline summarization/hf_transformers/model/run_summarization.py:1]: this file should be in the `scripts` folder

File: summarization/hf_transformers/model/run_summarization.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+python run_summarization.py \
```
Comment (SiddhantSadangi) [inline summarization/hf_transformers/model/run_summarization.sh:1]: this file should be in the `scripts` folder

File: summarization/hf_transformers/model/utils.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
+from functools import partial
```
Comment (SiddhantSadangi) [inline summarization/hf_transformers/model/utils.py:1]: this file should be in the `scripts` folder

File: summarization/hf_transformers/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+datasets >= 1.8.0
+sentencepiece != 0.1.92
+protobuf
+rouge-score
+nltk
+py7zr
+torch >= 1.3
+evaluate
+neptune-client
+transformers >= 4.22
```
Comment (SiddhantSadangi) [inline summarization/hf_transformers/requirements.txt:11]: lets pin this to the actual version you've used

File: summarization/hf_transformers/notebooks/project-metadata.ipynb [hunk 7]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (SiddhantSadangi) [inline summarization/hf_transformers/notebooks/project-metadata.ipynb:1]: I think we can remove this notebook. Doesn't really do much wrt the integration

**Review by SiddhantSadangi — CHANGES_REQUESTED** [review]
Left a few comments. We can rename the `model` folder to `scripts`

File: summarization/hf_transformers/requirements.txt (hunk 6 above)
Comment (kshitij12345) [reply summarization/hf_transformers/requirements.txt:11]: On looking a bit, the problem is we have used the development version (4.22 which is yet to be released).

Our integration will be released with that version. I tried to see if there is an easy way to install the nightly version of `transformers` via pip but there isn't. You have to manually install in locally.

File: summarization/hf_transformers/notebooks/project-metadata.ipynb (hunk 7 above)
Comment (kshitij12345) [reply summarization/hf_transformers/notebooks/project-metadata.ipynb]: Done!

**Review by SiddhantSadangi — APPROVED** [review]

**Pull request merged**